File: api/Models/StoreProduct.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace api.Models
{
    public class StoreProduct
    {
        public int StoreId { get; set; }
        public Store Store { get; set; } = default!;
        public int ProductId { get; set; }
        public Product Product { get; set; } = default!;
        [Required]
        [Precision(18, 2)]
        public decimal BuyPrice { get; set; }
        [Required]
        [Precision(18, 2)]
        public decimal SellPrice { get; set; }
        [Required]
        [Precision(18, 2)]
        public decimal Igv { get; set; }
        [Required]
        [Precision(18, 2)]
        public decimal Profit { get; set; }
        [Required]
        public int Stock { get; set; }
        [Required]
        public bool Salable { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
    }
}

File: api/app/models/store_product.ts
import { DateTime } from 'luxon';
import { BaseModel, belongsTo, column } from '@adonisjs/lucid/orm';
import Store from './store.js';
import type { BelongsTo } from '@adonisjs/lucid/types/relations';
import Product from './product.js';

export default class StoreProduct extends BaseModel {
  @column()
  declare storeId: number;

  @column()
  declare productId: number;

  @column()
  declare buyPrice: number;

  @column()
  declare sellPrice: number;

  @column()
  declare igv: number;

  @column()
  declare profit: number;

  @column()
  declare stock: number;

  @column()
  declare salable: boolean;

  @column.dateTime({ autoCreate: true })
  declare createdAt: DateTime;

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  declare updatedAt: DateTime;

  @column.dateTime()
  declare deletedAt: DateTime | null;

  @belongsTo(() => Store)
  declare store: BelongsTo<typeof Store>;

  @belongsTo(() => Product)
  declare product: BelongsTo<typeof Product>;
}
